lib/punycode.js

```diff
@@ -82,7 +82,7 @@ function map(array, callback) {
  * function.
  */
 function mapDomain(domain, callback) {
-	const parts = domain.split('@');
+	const parts = domain.split('@', 2);
```

Comment:
Same here, it's deprecated, let's not make changes to the implementation
```suggestion
	const parts = domain.split('@');
```